src/librustc_mir/borrow_check/nll/type_check/mod.rs

```diff
@@ -970,15 +970,21 @@ impl<'a, 'gcx, 'tcx> TypeChecker<'a, 'gcx, 'tcx> {
         locations: Locations,
         category: ConstraintCategory,
     ) -> Fallible<()> {
-        relate_tys::relate_type_and_user_type(
+        let ty = relate_tys::relate_type_and_user_type(
```

Comment:
It is okay to have a variable name the same as the module name ?

Reply:
yes